lib/fast_jsonapi/object_serializer.rb

```diff
@@ -176,9 +175,23 @@ def set_id(id_name = nil, &block)
       end
 
       def cache_options(cache_options)
-        self.cached = cache_options[:enabled] || false
-        self.cache_length = cache_options[:cache_length] || 5.minutes
-        self.race_condition_ttl = cache_options[:race_condition_ttl] || 5.seconds
+        cache_options = cache_options.dup
+
+        if cache_options.key?(:store)
+          self.cache_store_instance = cache_options.delete(:store)
+        elsif !cache_store_instance
```

Comment:
@doits why do we need an `elsif` here? Looks like `else` is just enough...

Reply:
It is there in case of inheritance. There might be a store already from parent, so we do not have to warn if users do this:

```rb
class BaseSerializer
  include FastJsonApi::ObjectSerializer

  cache_options store: MyStore.new, expires_in: 2.hours
end

class UserSerializer < BaseSerializer
  cache_options expires_in: 5.minutes
end
```

Reply:
As per https://github.com/fast-jsonapi/fast_jsonapi/pull/52#discussion_r372516561 this is not possible anymore.